src/rust/pal/functions.rs

```diff
@@ -0,0 +1,22 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+use crate::pal::data_structures::SockAddrIn;
+
+//======================================================================================================================
+// Windows functions
+//======================================================================================================================
+
+#[cfg(target_os = "windows")]
+pub fn get_addr_from_sock_addr_in(sock_addr_in: SockAddrIn) -> u32 {
```

Comment:
This is copying parameters in. Could you rely on a read-only reference parameter, please?

Reply:
Done